src/url-params.ts

```diff
@@ -0,0 +1,111 @@
+import * as _ from 'lodash';
+import {
+	getCookie,
+	setCookie,
+	hasCookieSupport,
+} from '@analytics/cookie-utils';
+import { Mixpanel } from 'mixpanel-browser';
+
+const URL_PARAM_DEVICE_ID = 'd_id';
+const COOKIES_DEVICE_IDS = '__analytics_dids';
+
+const deviceIdSeparator = /\s*,\s*/;
+
+const cookiesSupported = hasCookieSupport();
+
+/**
+ * AnalyticsUrlParams helps with handling analytics-related URL parameters
+ * like anonymous device identifier.
+ */
+export class AnalyticsUrlParams {
+	private deviceIds: string | null;
+
+	constructor(private mixpanel?: Mixpanel) {}
```

Comment:
How about you check here if there are deviceIds in cookies, and set the class `deviceIds` from there? Like that you can only reference the class property, and only the cookies when you are setting.

Reply:
Check the current impl.